pvlib/snow.py

```diff
@@ -0,0 +1,182 @@
+        Accumulated snowfall within each time period. [cm]
+    poa_irradiance : Series
+        Total in-plane irradiance [W/m^2]
+    temp_air : Series
+        Ambient air temperature [C]
+    surface_tilt : numeric
+        Tilt of module's from horizontal, e.g. surface facing up = 0,
+        surface facing horizon = 90. [degrees]
+    initial_coverage : float, default 0
+        Fraction of row's slant height that is covered with snow at the
+        beginning of the simulation. [unitless]
+    threshold_snowfall : float, default 1.0
+        Hourly snowfall above which snow coverage is set to the row's slant
+        height. [cm/hr]
+    can_slide_coefficient : float, default -80.
+        Coefficient to determine if snow can slide given irradiance and air
+        temperature. [W/(m^2 C)]
+    slide_amount_coefficient : float, default 0.197
+        Coefficient to determine fraction of snow that slides off in one hour.
+        [unitless]
+
+    Returns
+    -------
+    snow_coverage : Series
+        The fraction of the slant height of a row of modules that is covered
+        by snow at each time step.
+
+    Notes
+    -----
+    In [1]_, `can_slide_coefficient` is termed `m`, and the value of
+    `slide_amount_coefficient` is given in tenths of a module's slant height.
+
+    References
+    ----------
+    .. [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+       "Measured and modeled photovoltaic system energy losses from snow for
+       Colorado and Wisconsin locations." Solar Energy 97; pp.112-121.
+    .. [2] Ryberg, D; Freeman, J. (2017). "Integration, Validation, and
+       Application of a PV Snow Coverage Model in SAM" NREL Technical Report
+       NREL/TP-6A20-68705
+    '''
+
+    # find times with new snowfall
+    new_snowfall = fully_covered_nrel(snowfall, threshold_snowfall)
+
+    # set up output Series
+    snow_coverage = pd.Series(np.nan, index=poa_irradiance.index)
+
+    # determine amount that snow can slide in each timestep
+    can_slide = temp_air > poa_irradiance / can_slide_coefficient
+    slide_amt = slide_amount_coefficient * sind(surface_tilt) * \
+        _time_delta_in_hours(poa_irradiance.index)
+    slide_amt[~can_slide] = 0.
+    # don't slide during snow events
+    slide_amt[new_snowfall] = 0.
+    # don't slide in the interval preceding the snowfall data
+    slide_amt.iloc[0] = 0
+
+    # build time series of cumulative slide amounts
+    sliding_period_ID = new_snowfall.cumsum()
+    cumulative_sliding = slide_amt.groupby(sliding_period_ID).cumsum()
+
+    # set up time series of snow coverage without any sliding applied
+    snow_coverage[new_snowfall] = 1.0
+    if np.isnan(snow_coverage.iloc[0]):
+        snow_coverage.iloc[0] = initial_coverage
+    snow_coverage.ffill(inplace=True)
+    snow_coverage -= cumulative_sliding
+
+    # clean up periods where row is completely uncovered
+    return snow_coverage.clip(lower=0)
+
+
+def dc_loss_nrel(snow_coverage, num_strings):
```

Comment:
Does this function's docstring need a reference?

Reply:
Yes